src/app/pages/services/components/service-smb/service-smb.component.ts

```diff
@@ -1,367 +1,170 @@
+    admin_group: helptext.cifs_srv_admin_group_tooltip,
+    bindip: helptext.cifs_srv_bindip_tooltip,
+    smb_options: helptext.cifs_srv_smb_options_tooltip,
+    aapl_extensions: helptext.cifs_srv_aapl_extensions_tooltip,
+  };
+
+  readonly logLevelOptions$ = of([
+    { label: this.translate.instant('None'), value: LogLevel.None },
+    { label: this.translate.instant('Minimum'), value: LogLevel.Minimum },
+    { label: this.translate.instant('Normal'), value: LogLevel.Normal },
+    { label: this.translate.instant('Full'), value: LogLevel.Full },
+    { label: this.translate.instant('Debug'), value: LogLevel.Debug },
+  ]);
+  readonly unixCharsetOptions$ = this.ws.call('smb.unixcharset_choices').pipe(choicesToOptions());
+  readonly guestAccountOptions$ = this.ws.call('user.query').pipe(
+    map((users) => users.map((user) => ({ label: user.username, value: user.username }))),
+  );
+  readonly adminGroupProvider = new SimpleAsyncComboboxProvider(
+    this.userService.groupQueryDsCache('', true).pipe(
+      map((groups) => groups.map((group) => ({ label: group.group, value: group.group }))),
+    ),
+  );
+  readonly bindIpAddressOptions$ = this.ws.call('smb.bindip_choices').pipe(choicesToOptions());
 
   constructor(
-    protected router: Router,
-    protected route: ActivatedRoute,
-    protected ws: WebSocketService,
-    protected servicesService: ServicesService,
-    protected idmapService: IdmapService,
-    protected userService: UserService,
-    protected loader: AppLoaderService,
-    protected dialog: MatDialog,
-  ) {}
-
-  resourceTransformIncomingRestData(data: SmbConfig): SmbConfig {
-    // If validIps is slow to load, skip check on load (It's still done on save)
-    if (this.validBindIps && Object.keys(this.validBindIps).length !== 0) {
-      return this.compareBindIps(data);
-    }
-    return data;
+    private ws: WebSocketService,
+    private errorHandler: FormErrorHandlerService,
+    private cdr: ChangeDetectorRef,
+    private fb: FormBuilder,
+    private router: Router,
+    private dialogService: DialogService,
+    private translate: TranslateService,
+    private userService: UserService,
+    private validatorsService: IxValidatorsService,
+  ) {
+    this.form.get('netbiosname_b').disable();
   }
 
-  compareBindIps(data: SmbConfig): SmbConfig {
-    // Weeds out invalid addresses (ie, ones that have changed). Called on load and on save.
-    data.bindip = data.bindip ? data.bindip : [];
-    if (this.validBindIps && Object.keys(this.validBindIps).length !== 0) {
-      data.bindip.forEach((ip) => {
-        if (!Object.values(this.validBindIps).includes(ip)) {
-          data.bindip.splice((data.bindip as any)[ip], 1);
-        }
-      });
-    } else {
-      data.bindip = [];
+  ngOnInit(): void {
+    this.isFormLoading = true;
+    if (window?.localStorage?.getItem('product_type')?.includes(ProductType.Enterprise)) {
+      this.form.get('netbiosname_b').disabledWhile(
```

Comment:
Also, not sure but we might able to simplify this logic by adding  `[disabled]="!hasSecondHandler"` in the template

Reply:
> Also, not sure but we might able to simplify this logic by adding `[disabled]="!hasSecondHandler"` in the template

```
It looks like you're using the disabled attribute with a reactive form directive. If you set disabled to true
  when you set up this control in your component class, the disabled attribute will actually be set in the DOM for
  you. We recommend using this approach to avoid 'changed after checked' errors.
  
  Example:
  form = new FormGroup({
    first: new FormControl({value: 'Nancy', disabled: true}, Validators.required),
    last: new FormControl('Drew', Validators.required)
  });
```
The proposed approach does not work due to type checking in form builder. Other ideas are welcomed